app/server/erlang/pi_server_api.erl

```diff
@@ -68,101 +72,151 @@
 %%   cancel its current timers.
 
 
-start(Port, CuePid) ->
-    Parent = self(),
-    Pid = spawn(fun() -> init(Parent, Port, CuePid) end),
-    register(?SERVER, Pid),
-    receive
-        {Pid, started} ->
-            Pid
-    end.
+%% supervisor compliant start function
+start_link() ->
+    %% synchronous start of the child process
+    proc_lib:start_link(?MODULE, init, [self()]).
 
 
-init(Parent, Port, CuePid) ->
-    {ok, APISocket} = gen_udp:open(Port, [binary, {ip, loopback}]),
-    Parent ! {self(), started},
-    TagMap = #{},
-    debug(2, "listening for API commands on socket: ~p~n",
-          [try erlang:port_info(APISocket) catch _:_ -> undefined end]),
-    loop(APISocket, 1, TagMap, CuePid).
+init(Parent) ->
+    register(?SERVER, self()),
+    APIPort = application:get_env(?APPLICATION, api_port, undefined),
+    io:format("~n"
+              "+--------------------------------------+~n"
+              "    This is the Sonic Pi API Server     ~n"
+              "       Powered by Erlang ~s             ~n"
+              "                                        ~n"
+              "       API listening on port ~p         ~n"
+              "+--------------------------------------+~n~n~n",
+              [erlang:system_info(otp_release), APIPort]),
 
+    {ok, APISocket} = gen_udp:open(APIPort, [binary, {ip, loopback}]),
 
-loop(APISocket, N, TagMap, CuePid) ->
+    %% tell parent we have allocated resources and are up and running
+    proc_lib:init_ack(Parent, {ok, self()}),
+
+    debug(2, "listening for API commands on socket: ~p~n",
+          [try erlang:port_info(APISocket) catch _:_ -> undefined end]),
+    State = #{parent => Parent,
+              api_socket => APISocket,
+              cue_server => pi_server_cue:server_name(),
```

Comment:
If you add the cue server_name as a start argument this module do not need to be aware of the pi_server_cue module name at all.

Reply:
Yes, that might be nicer.

Reply:
Added.